test/integration/scheduler/scheduler_test.go

```diff
@@ -162,39 +162,34 @@ var _ = ginkgo.Describe("Scheduler", func() {
 		gomega.Expect(createdProdJob2.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
 	})
 
-	ginkgo.It("Should schedule jobs on tolerated flavors", func() {
-		ginkgo.By("checking a job without toleration starts on the non-tainted flavor")
-		job1 := testing.MakeJob("on-demand-job1", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "5").Obj()
-		gomega.Expect(k8sClient.Create(ctx, job1)).Should(gomega.Succeed())
-		createdJob1 := &batchv1.Job{}
-		gomega.Eventually(func() *bool {
-			lookupKey := types.NamespacedName{Name: job1.Name, Namespace: job1.Namespace}
-			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob1)).Should(gomega.Succeed())
-			return createdJob1.Spec.Suspend
-		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
-		gomega.Expect(createdJob1.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+	ginkgo.It("Should schedule workloads on tolerated flavors", func() {
+		ginkgo.By("checking a workload without toleration starts on the non-tainted flavor")
+		wl1 := testing.MakeWorkload("on-demand-wl1", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "5").Obj()
+		gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
 
-		// TODO(#8): uncomment the following once we have proper re-queueing.
-		// ginkgo.By("checking a second job without toleration doesn't start")
-		// job2 := testing.MakeJob("on-demand-job2", namespace).Queue(queue.Name).Request(corev1.ResourceCPU, "5").Obj()
-		// gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
-		// createdJob2 := &batchv1.Job{}
-		// gomega.Consistently(func() bool {
-		// 	lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
-		// 	return k8sClient.Get(ctx, lookupKey, createdJob2) == nil && *createdJob2.Spec.Suspend
-		// }, consistentDuration, interval).Should(gomega.BeTrue())
-
-		ginkgo.By("checking a third job with toleration starts")
-		job3 := testing.MakeJob("spot-job3", ns.Name).Queue(prodQueue.Name).
-			Toleration(spotToleration).
-			Request(corev1.ResourceCPU, "5").Obj()
-		gomega.Expect(k8sClient.Create(ctx, job3)).Should(gomega.Succeed())
-		createdJob3 := &batchv1.Job{}
-		gomega.Eventually(func() *bool {
-			gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: job3.Name, Namespace: job3.Namespace}, createdJob3)).Should(gomega.Succeed())
-			return createdJob3.Spec.Suspend
-		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
-		gomega.Expect(createdJob3.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotTaintedFlavor.Name))
+		wlCopy := &kueue.Workload{}
+		gomega.Eventually(func() bool {
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl1), wlCopy)).To(gomega.Succeed())
+			return wlCopy.Spec.Admission != nil
```

Comment:
```suggestion
			return wlCopy.Spec.Admission
```

Reply:
Hi Aldo Did you suggest to change it like this?
```go
expectAdmission := testing.MakeAdmission(prodClusterQ.Name).Flavor(corev1.ResourceCPU,onDemandFlavor.Name).Obj()
gomega.Eventually(func() kueue.Admission {
    gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl1), wlCopy)).To(gomega.Succeed())
    return wlCopy.Spec.Admission or *wlCopy.Spec.Admission
}, framework.Timeout, framework.Interval).Should(gomega.Equal(expectAdmission))
```


`wlCopy.Spec.Admission` is a point of `kueue.Admission`. We still need to check if it's nil before getting value of it. And we'd better not continuity to check the value of `Admission`, we just need to check it when the first time it's not nil.

Reply:
Oops. My recommendation is to return `*kueue.Admission`.

`gomega.Equal` and `pkg/util/testing.Equal` should be able to handle it.